### PR TITLE
Add method to Sequence to estimate delay introduced before adding pulse

### LINKED ISSUE #761: Function to estimate delay needed between two pulses
I was wondering if it would be possible to have a function in `Sequence` (or `Channel` or anywhere convenient) to estimate the delay needed between two pulses on a physical device?

Context: I want to simulate the dynamics for a global pulse for the maximum duration allowed by `AnalogDevice`. In some cases, I want to start the dynamics with a "short" pulse to change the initial state. To stay below `device.max_sequence_duration`, I want to reduce the duration of the long pulse. This is easy for `MockDevice` since the pulses are sharp, but the pulses are modified for `AnalogDevice` to respect the physical constraints of the device. In particular, a delay can be added between the two pulses. Since I cannot get the duration of the delay before adding the long pulse to the sequence, but `seq.add()` raises an error because maximal duration is exceeded, I am stuck:
```
RuntimeError: The sequence's duration exceeded the maximum duration allowed by the device (4000 ns).
```

Here is the relevant part of my code:
```python
device = pulser.AnalogDevice
channel = seq.declared_channels["ising"]
seq = pulser.Sequence(register, device)
seq.declare_channel("ising", "rydberg_global")

pi2_wf = pulser.BlackmanWaveform.from_max_val(channel.max_amp, np.pi / 2)
hadamard = pulser.Pulse.ConstantDetuning(
    amplitude=pi2_wf, detuning=0, phase=np.pi / 2, post_phase_shift=np.pi
)
seq.add(hadamard, "ising")

duration = 4000
cst_pulse = pulser.Pulse.ConstantPulse(
        duration=duration, amplitude=Omega, detuning=delta, phase=0.
)
# this add delay but also raises error
seq.add(hadamard, "ising")
```

I tried to investigate the source, but did not manage to extract a simple function to get the delay. For example, I thought that this would work:
```python
state_prep_duration = hadamard.duration
delay = seq._schedule._find_add_delay(state_prep_duration, channel="ising", protocol="min-delay")
```
but it does not give the delay observed inside the full sequence (which I get by just decreasing duration until it works, which is not generalizable):
```
Channel: ising
t: 0 | Initial targets: 0, 1, 2, 3 | Phase Reference: 0.0 
t: 0->500 | Pulse(Amp=Blackman(Area: 1.57) rad/µs, Detuning=0 rad/µs, Phase=1.57) | Targets: 0, 1, 2, 3
t: 500->696 | Delay 
t: 696->3956 | Pulse(Amp=5 rad/µs, Detuning=5 rad/µs, Phase=3.14) | Targets: 0, 1, 2, 3
```

### PR DESCRIPTION
**Pull request by a-corni**
Adds the method `estimate_add_delay` to Sequence.
Has the same signature as the `add` method.
Takes into account AOM and EOM Pulses (raise a Warning if the Pulse added is not Constant, with amplitude and detuning equal to the rabi_fres and detuning_on of the EOM Block)
Raise a Warning if the duration of the added Pulse exceeds the duration of the Sequence.
Fixes #761

### CODE REVIEW

**Review by HGSilveri — COMMENTED** [review]
Looks good, I only have nits :)

File: pulser-core/pulser/sequence/_schedule.py [hunk 1]
```diff
@@ -396,14 +396,15 @@ def disable_eom(self, channel_id: str, _skip_buffer: bool = False) -> None:
...
-    def add_pulse(
+    def get_next_time_slot(
         self,
         pulse: Pulse,
         channel: str,
         phase_barrier_ts: list[int],
         protocol: str,
         phase_drift_params: _PhaseDriftParams | None = None,
-    ) -> None:
+        block_if_exceeds: bool = False,
```
[the remark this answers is not in the document]
Comment (a-corni) [reply pulser-core/pulser/sequence/_schedule.py:406]: It's already better ;)

File: tests/test_sequence.py [hunk 2]
```diff
@@ -1729,6 +1729,95 @@ def test_sequence(reg, device, patch_plt_show):
...
+            match="Channel ising is in EOM mode, detuning of the pulse",
+        ):
+            assert (
+                seq.estimate_added_delay(
+                    Pulse.ConstantPulse(100, 1, 1, 0), "ising"
+                )
+                == 0
+            )
+    assert seq.estimate_added_delay(pulse_0, "ising", "min-delay") == 0
+    seq._add(pulse_0, "ising", "min-delay")
```
[the remark this answers is not in the document]
Comment (a-corni) [reply tests/test_sequence.py:1768]: I can't do `add(pulse_0, "ising")` when "ising" is in EOM mode. I can have `seq.add(pulse_0, "ising") if not eom else seq.add_eom_pulse("ising", 100, 0)` but that would have the same outcome

File: tests/test_sequence.py [hunk 3]
```diff
@@ -1729,6 +1729,95 @@ def test_sequence(reg, device, patch_plt_show):
...
+    seq.add(pulse_0, "ising")
+    assert seq.estimate_added_delay(det_pulse, "dmm_0") == 0
+    with pytest.raises(
+        ValueError, match="The detuning in a DMM must not be positive."
+    ):
+        seq.estimate_added_delay(Pulse.ConstantPulse(100, 0, 1, 0), "dmm_0")
+    with pytest.raises(
+        ValueError, match="The pulse's amplitude goes over the maximum"
+    ):
+        seq.estimate_added_delay(pulse_0, "dmm_0")
```
[the remark this answers is not in the document]
Comment (a-corni) [reply tests/test_sequence.py:1818]: Yes that's True, I hesitated ;)

**Review by HGSilveri — APPROVED** [review]
LGTM!

**Pull request merged**